pytest_mock.py

```diff
@@ -180,7 +181,9 @@ def assert_wrapper(__wrapped_mock_method__, *args, **kwargs):
             actual_args, actual_kwargs = __mock_self.call_args
             assert actual_args == args[1:]
             assert actual_kwargs == kwargs
-        raise AssertionError(*e.args)
+        err = e
+    if err is not None:
+        raise AssertionError(*err.args)
```

Comment:
@blueyed yeah this is constructed this way because if the assertion is raised inside the
`except` block then you still get the `during the above exception` message/tb. If the `e` isn't assigned to some other variable then it is a NameError in py3. If this project didn't support py2 we could theoretically do `raise ... from None` but that's illegal before py3 _and_ it didn't actually work correctly for me in my testing.